deduce/annotate.py

```diff
@@ -428,12 +428,18 @@ def annotate_institution(text):
     # Return the text
     return text
 
+def get_date_replacement(date_match: re.Match) -> str:
+    punctuation = date_match['n']
```

Comment:
What happens if you index a re.Match with 'n'? Could not find it here: https://docs.python.org/3/library/re.html#match-objects

Reply:
Indexing a Match object is equivalent to calling the group method. Thus:
aMatch['n']
is the same as
aMatch.group('n')

In this case, the group 'n' was defined in the Regex pattern. I have made this clearer with documentation and variable naming